source/common/stats/raw_stat_data.cc

```diff
@@ -24,6 +24,8 @@ uint64_t roundUpMultipleNaturalAlignment(uint64_t val) {
 
 } // namespace
 
+RawStatDataAllocator::~RawStatDataAllocator() {}
```

Comment:
Up to you and @fredlas how you want to reconcile this PR and the hot restart PR. The hot restart PR is very close to merging but I will let both of you decide who goes first.